x-pack/plugin/esql/qa/testFixtures/src/main/resources/inlinestats.csv-spec

```diff
@@ -1477,3 +1467,932 @@ ROW salary = 12300, emp_no = 5, gender = "F"
+| sort emp_no desc
+| keep emp_no, salary, max, min, count*, gender, languages
+| limit 5
+;
+
+    emp_no:i   |    salary:i   |      max:d    |      min:d    |     count:l   |count_distinct:l|    gender:s  |languages:i   
+10100          |68431          |null           |null           |0              |0               |F              |4              
+10099          |73578          |null           |null           |0              |0               |F              |2              
+10098          |44817          |null           |null           |0              |0               |F              |4              
+10097          |71165          |null           |null           |0              |0               |M              |3              
+10096          |43889          |null           |null           |0              |0               |M              |4              
+;
+
+prunedInlinestatsFollowedByInlinestats_GroupByOneFieldEach_DifferentFields
+required_capability: inlinestats_v10
+from employees
+| eval my_length = length(concat(first_name, null))
+| inlinestats count = count(my_length) where false,
+        values = mv_slice(values(first_name), 0, 1) where my_length > 0 by languages
+| keep emp_no, first_name, *length, count, values, languages, gender
+| inlinestats count_distinct = count_distinct(count) by gender
+| sort emp_no
+| limit 3
+;
+
+    emp_no:i   |  first_name:s |   my_length:i |     count:l   |    values:s   |   languages:i |count_distinct:l|    gender:s     
+10001          |Georgi         |null           |0              |null           |2              |1               |M              
+10002          |Bezalel        |null           |0              |null           |5              |1               |F              
+10003          |Parto          |null           |0              |null           |4              |1               |M              
+;
+
+prunedInlinestatsFollowedByInlinestats_GroupByOneFieldEach_SameFields
+required_capability: inlinestats_v10
+from employees 
+| eval my_length = length(concat(first_name, null)) 
+| inlinestats count = count(my_length) where false,
+        values = mv_slice(values(first_name), 0, 1) where my_length > 0 by languages
+| keep emp_no, first_name, *length, count, values, languages, gender
+| inlinestats count_distinct = count_distinct(count) by languages
+| sort emp_no
+| limit 3
+;
+
+    emp_no:i   |  first_name:s |   my_length:i |     count:l   |    values:s   |   gender:s    |count_distinct:l|   languages:i   
+10001          |Georgi         |null           |0              |null           |M              |1               |2              
+10002          |Bezalel        |null           |0              |null           |F              |1               |5              
+10003          |Parto          |null           |0              |null           |M              |1               |4             
+;
+
+prunedInlinestatsFollowedByInlinestats_GroupByOneFieldOnSecondInlinestats-Ignore
```

Comment:
This will be as a follow up investigation. I did try to fix it several times, ran out of good ideas.